contracts/Elections.sol

```diff
@@ -273,24 +299,33 @@ contract Elections is IElections, ContractRegistryAccessor {
-		_applyDelegatedStake(delegatee, newUncappedStake);
+	function notifyStakeChange(uint256 prevDelegateTotalStake, uint256 newDelegateTotalStake, address delegate, bool isSelfDelegatingDelegate) external onlyDelegationsContract {
 
-		_applyStakesToBanningBy(delegatee, prevGovStakeDelegatee);
+		uint256 prevGovStakeDelegate = getGovernanceEffectiveStake(isSelfDelegatingDelegate, prevDelegateTotalStake);
+		uint256 newGovStakeDelegate = getGovernanceEffectiveStake(isSelfDelegatingDelegate, newDelegateTotalStake);
+
+		totalGovernanceStake = totalGovernanceStake.sub(prevGovStakeDelegate).add(newGovStakeDelegate);
+
+		_applyDelegatedStake(delegate, newDelegateTotalStake);
+
+		_applyStakesToBanningBy(delegate, prevGovStakeDelegate);
 	}
 
-	function notifyStakeChangeBatch(address[] calldata stakeOwners, uint256[] calldata newUncappedStakes, uint256[] calldata prevGovStakeOwners, address[] calldata delegatees, uint256[] calldata prevGovStakeDelegatees) external onlyDelegationsContract {
-		require(stakeOwners.length == newUncappedStakes.length, "arrays must be of same length");
-		require(stakeOwners.length == prevGovStakeOwners.length, "arrays must be of same length");
-		require(stakeOwners.length == delegatees.length, "arrays must be of same length");
-		require(stakeOwners.length == prevGovStakeDelegatees.length, "arrays must be of same length");
+	function notifyStakeChangeBatch(uint256[] calldata prevDelegateTotalStakes, uint256[] calldata newDelegateTotalStakes, address[] calldata delegates, bool[] calldata isSelfDelegatingDelegates) external onlyDelegationsContract {
+		require(prevDelegateTotalStakes.length == newDelegateTotalStakes.length, "arrays must be of same length");
+		require(prevDelegateTotalStakes.length == delegates.length, "arrays must be of same length");
+		require(prevDelegateTotalStakes.length == isSelfDelegatingDelegates.length, "arrays must be of same length");
+
+		for (uint i = 0; i < prevDelegateTotalStakes.length; i++) {
+			uint256 prevGovStakeDelegate = getGovernanceEffectiveStake(isSelfDelegatingDelegates[i], prevDelegateTotalStakes[i]);
+			uint256 newGovStakeDelegate = getGovernanceEffectiveStake(isSelfDelegatingDelegates[i], newDelegateTotalStakes[i]);
 
-		for (uint i = 0; i < stakeOwners.length; i++) {
+			totalGovernanceStake = totalGovernanceStake.sub(prevGovStakeDelegate).add(newGovStakeDelegate);
 
-			// this mimics notifyStakeChange. TODO optimize to minimize calls to committe contract assuming similar delegatees are consecutive in order. careful not to break banning logic...
-			_applyDelegatedStake(delegatees[i], newUncappedStakes[i]);
+			_applyDelegatedStake(delegates[i], newDelegateTotalStakes[i]);
 
-			_applyStakesToBanningBy(delegatees[i], prevGovStakeDelegatees[i]);
+			// TODO add tests to show banning votes are evaluated equally when stake change is batched and not batched
```

Comment:
Not sure that's a hard requirement as long as it is consistent with banning logic, especially if we can avoid state writes inside a loop.